packages/cli-tools/bin/streamr-keys-generate.ts

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env node
+import '../src/logLevel'
+import { KeyPairIdentity } from '@streamr/sdk/dist/types/src/identity/KeyPairIdentity'
+import { identityConfig, validKeyTypeValues } from '@streamr/sdk'
+import { binaryToHex } from '@streamr/utils'
+import { createCommand, Options } from '../src/command'
+
+createCommand()
+    .description('generate a public/private key pair based on the --key-type option')
+    .option('--key-type [key-type]', `one of: [${validKeyTypeValues.join(', ')}]`)
+    .action(async (options: Options) => {
+        if (!options.keyType) {
+            console.error(`Error: Please provide --key-type [one of: ${validKeyTypeValues.join(', ')}]`)
+        } else {
+            const config = identityConfig[options.keyType]
+            if (!config) {
+                console.error(`Error: Invalid key type. Must be one of: ${validKeyTypeValues.join(', ')}.`)
+            }
+            const identity = config.generate() as KeyPairIdentity
+            console.info(`Public key: ${await identity.getUserIdString()}`)
```

Comment:
Currently a `KeyPairIdentity` consists of `userId` + `privateKey`, not `publicKey` + `privateKey`. As `userId` and publicKey are not the same in `Ethereum`, it is good to keep this seperation.

But whenever we output a `userId` it would be good to label it as user ID, not public key.
```suggestion
            console.info(`User ID: ${await identity.getUserIdString()}`)
```

Reply:
Somewhat disagree as mentioned above. First of all, in Ethereum terminology, there is no such thing as user ID. There is an 'address', which is derived from the public key by hashing it. For all intents and purposes, the address serves the same function as a public key, it's just slightly compressed.

A key pair consists of a public key and a private key, regardless of whether we (or Ethereum) give weird names like `address` or `userId` to them.

Especially in the bigger picture where Streamr supports many different types of key pairs (all of which call the public key the public key, except the Ethereum one), I would prefer to use the established and more common terminology.

Reply:
We talked about this and agreed that `UserID` is on a higher abstraction level compared to public key. `KeyPairIdentity` - consisting of a public key and private key - is defined as an `Identity` that uses the public key as the user ID. Therefore it's exactly right to:
- On the `Identity` level, use the highest abstraction level (define getters like `getUserId()`)
- On the `KeyPairIdentity` level, internally it's clearest to use `this.publicKey` and `this.privateKey` since those concepts are closest to the definition of what a `KeyPairIdentity` is, and `KeyPairIdentity` uses signing functions etc. that use the public key / private key terminology, so it's best to maintain naming correspondence with them.